src/controllers/routeActivation.ts

```diff
@@ -0,0 +1,148 @@
+import { Context } from 'koa';
+import dotenv from 'dotenv';
+import { ProcessorTransformationRequest, RouterTransformationRequest } from '../types';
+
+dotenv.config();
+
+const transformerMode = process.env.TRANSFORMER_MODE;
+const startDestTransformer = transformerMode === 'destination' || !transformerMode;
+const startSourceTransformer = transformerMode === 'source' || !transformerMode;
+const transformerDelivery = process.env.TRANSFORMER_DELIVERY || true;
+const deliveryTestModeEnabled =
+  process.env.TRANSFORMER_DELIVERY_TEST_ENABLED?.toLowerCase() === 'true' || false;
+const transformerTestModeEnabled = process.env.TRANSFORMER_TEST_MODE
+  ? process.env.TRANSFORMER_TEST_MODE.toLowerCase() === 'true'
+  : false;
+const areFunctionsEnabled = process.env.ENABLE_FUNCTIONS === 'false' ? false : true;
+const destinationFilterList = process.env.DESTINATION_FILTER_LIST?.toLocaleLowerCase();
+const sourceFilteList = process.env.SOURCE_FILTER_LIST?.toLocaleLowerCase();
+const deliveryFilterList = process.env.DESTINATION_DELIVERY_FILTER_LIST?.toLocaleLowerCase();
+
+export default class RouteActivationController {
+  public static isDestinationRouteActive(ctx: Context, next: any) {
+    if (!startDestTransformer) {
```

Comment:
all the below functions can be written using one helper function.

```js
function isRouteActive(condition: boolean, ctx: Context, next: any) {
    if (!condition) {
      ctx.status = 404;
      ctx.body = 'This route is disabled'
      return ctx;
    } else {
      return next();
    }
}
```

Reply:
handled